### PR TITLE
Update oxygen-xml-editor.rb

### PR DESCRIPTION
**Pull request by bwbohl**
add sha256 sum

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]

**Comment by bwbohl** [comment]
* Unfortunately `brew audit --cask --online https://raw.githubusercontent.com/bwbohl/homebrew-cask/patch-1/Casks/oxygen-xml-editor.rb`reports error: `Use `sha256 :no_check` when URL is unversioned.`
* `brew style --fix https://raw.githubusercontent.com/bwbohl/homebrew-cask/patch-1/Casks/oxygen-xml-editor.rb` reports 5 offenses that could be fixed:
  

- [ ] C: [Corrected] Magic comments should be in the following order: encoding, typed, warn_indent, frozen_string_literal.
  ```bash
    # frozen_string_literal: true
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- [ ] C: [Corrected] No Sorbet sigil found in file. Try a typed: false to start (you can also use rubocop -a to automatically add this).
  ```bash
  cask "oxygen-xml-editor" do
  ^^^^
  ```
- [ ] C: [Corrected] Missing frozen string literal comment.
  ```bash 
    cask "oxygen-xml-editor" do
    ^
  ```
- [ ] C: [Corrected] Magic comments should be in the following order: encoding, typed, warn_indent, frozen_string_literal.
  ```bash
  # typed: false
  ^^^^^^^^^^^^^^
  ```
- [ ] C: [Corrected] Add an empty line after magic comments.
  ```bash
  cask "oxygen-xml-editor" do
  ^
  ```

File: Casks/oxygen-xml-editor.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cask "oxygen-xml-editor" do
   version "25.0,2022121306"
-  sha256 :no_check
+  sha256 "6dbb317eafbb9c450608629fdeae0cdd6ca2f84e1284c7573a2090488d9aa14b"
 
   url "https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen-openjdk.dmg"
```
Comment (bwbohl) [inline Casks/oxygen-xml-editor.rb:5]:
```suggestion
  url "https://archives.oxygenxml.com/Oxygen/Editor/InstData#{version.split(',').first}/MacOSX/VM/oxygen-openjdk.dmg"
```

File: Casks/oxygen-xml-editor.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 cask "oxygen-xml-editor" do
   version "25.0,2022121306"
-  sha256 :no_check
+  sha256 "6dbb317eafbb9c450608629fdeae0cdd6ca2f84e1284c7573a2090488d9aa14b"
 
-  url "https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen-openjdk.dmg"
+  url "https://archives.oxygenxml.com/Oxygen/Editor/InstData#{version.split(',').first}/MacOSX/VM/oxygen-openjdk.dmg"
```
Comment (bwbohl) [inline Casks/oxygen-xml-editor.rb:5]:
```suggestion
  url "https://archives.oxygenxml.com/Oxygen/Editor/InstData#{version.split(",").first}/MacOSX/VM/oxygen-openjdk.dmg"
```

**Comment by SMillerDev** [comment]
You should run `brew audit --cask --online oxygen-xml-editor` though.

File: Casks/oxygen-xml-editor.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 cask "oxygen-xml-editor" do
   version "25.0,2022121306"
-  sha256 :no_check
+  sha256 "6dbb317eafbb9c450608629fdeae0cdd6ca2f84e1284c7573a2090488d9aa14b"
 
-  url "https://www.oxygenxml.com/InstData/Editor/MacOSX/VM/oxygen-openjdk.dmg"
+  url "https://archives.oxygenxml.com/Oxygen/Editor/InstData#{version.split(",").first}/MacOSX/VM/oxygen-openjdk.dmg"
```
Comment (SMillerDev) [inline Casks/oxygen-xml-editor.rb:5]:
```suggestion
  url "https://archives.oxygenxml.com/Oxygen/Editor/InstData#{version.csv.first}/MacOSX/VM/oxygen-openjdk.dmg"
```

Comment (bevanjkay) [reply Casks/oxygen-xml-editor.rb:5]: @bwbohl The maintainers preference is what @SMillerDev suggested.

**Review by SMillerDev — APPROVED** [review]

**Comment by miccal** [comment]
Thank you @bwbohl.

File: Casks/oxygen-xml-editor.rb (hunk 3 above)
Comment (bwbohl) [reply Casks/oxygen-xml-editor.rb]: @SMillerDev sorry for not having identified you as member of homebrew 😅

**Pull request merged**